src/tests/posix/pthread/pthread_rwlock_test.c

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file
+ *
+ * @date Mar 02, 2020
+ * @author: Puranjay Mohan
+ */
+
+#include <pthread.h>
+#include <poll.h>
+#include <embox/test.h>
+#include <stdlib.h>
+#define ACCESS_ONCE(x) (*(volatile typeof(x) *) &(x))
+
+EMBOX_TEST_SUITE("posix/pthread_rwlock_api");
+
+struct data {
+	pthread_rwlock_t lock_rw;
+	int x;
+	int reader_x;
+};
+
+TEST_CASE ("Initialize rwlock with pthread_rwlock_init()") {
+
+	pthread_rwlock_t rw;
+	test_assert_zero(pthread_rwlock_init(&rw, NULL));
+	test_assert_zero(pthread_rwlock_rdlock(&rw));
+	//function pthread_rwlock_trywrlock() is not implemented hence returns ENOSYS
+	test_assert_equal(pthread_rwlock_trywrlock(&rw), -ENOSYS);
+	test_assert_zero(pthread_rwlock_unlock(&rw));
+	test_assert_zero(pthread_rwlock_wrlock(&rw));
+	//function pthread_rwlock_tryrdlock() is not implemented hence returns ENOSYS
+	test_assert_equal(pthread_rwlock_tryrdlock(&rw), -ENOSYS);
+	test_assert_zero(pthread_rwlock_unlock(&rw));
+	test_assert_zero(pthread_rwlock_destroy(&rw));
+}
+
+void *reader_thread(void *arg) {
+
+	int i;
+	int newx, oldx;
+	newx = oldx = -1;
```

Comment:
please don't mix variables declaration and logic variables

Reply:
corrected